app/controllers/account/personal_details_controller.rb

```diff
@@ -0,0 +1,39 @@
+class Account::PersonalDetailsController < ApplicationController
+  layout "admin_layout"
+
+  before_action :authenticate_user!
+  before_action :authorise_user
+
+  def show; end
+
+  def update_organisation
+    new_organisation_id = params[:user][:organisation_id]
+    new_organisation = Organisation.find(new_organisation_id)
+
+    if current_user.update(organisation_id: new_organisation_id)
+      redirect_to account_path, notice: "Your organisation is now #{new_organisation.name}"
+    else
+      flash[:alert] = "There was a problem changing your organisation."
+      render :show
+    end
+  end
+
+  def update_role
+    previous_role = current_user.role
+    new_role = params[:user][:role]
+
+    if current_user.update(role: new_role)
```

Comment:
I think we should probably use the `UserUpdate` class here, so that we also create the relevant `EventLog`s (see https://github.com/alphagov/signon/blob/main/app/services/user_update.rb#L78).